platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java

```diff
@@ -90,8 +93,12 @@ public static void handleNetworkException(final Exception e, final Connection co
         // we use a different marker depending on what the root cause is
         final Marker marker = NetworkUtils.determineExceptionMarker(e);
         if (SOCKET_EXCEPTIONS.getMarker().equals(marker)) {
-            final String formattedException = NetworkUtils.formatException(e);
-            logger.warn(marker, "Connection broken: {} {}", description, formattedException);
+            if (socketExceptionRateLimiter.requestAndTrigger()) {
+                logger.warn(marker, "Connection broken: {}", description, e);
```

Comment:
not in a compiler so can't check but I think this should be `Connection broken: {} {}` so the placeholders match the args?

Reply:
No. This is a feature of the Logger interface - it accepts one extra argument of type `Throwable`, and when present, it prints its entire stack trace. If there was an explicit `{}` in the formatting string, then the logger would only print the results of the `throwable.toString()` which wouldn't include the stack trace and which would defeat the purpose of this very fix.

This is how you usually want to log all your exceptions, unless they are too noisy like this one, where we have to do it only occasionally using the RateLimiter.

So this code is written by design and there's not a bug w.r.t. the `e`/`{}`.